arch/x86/core/pcie.c

```diff
@@ -206,22 +206,19 @@ uint32_t pcie_msi_map(unsigned int irq,
 uint16_t pcie_msi_mdr(unsigned int irq,
 		      msi_vector_t *vector)
 {
-#ifdef CONFIG_PCIE_MSI_X
-	if ((vector != NULL) && (vector->msix)) {
-		return 0x4000U | vector->arch.vector;
-	}
-#endif
+	uint16_t msg_data_reg = 0;
 
-	if (vector == NULL) {
+#if !defined(CONFIG_INTEL_VTD_ICTL)
+	if (vector != NULL) {
+		msg_data_reg = vector->arch.vector;
+	} else {
 		/* edge triggered */
-		return 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
+		msg_data_reg = 0x4000U | Z_IRQ_TO_INTERRUPT_VECTOR(irq);
 	}
-
-	/* VT-D requires it to be 0, so let's return 0 by default */
-	return 0;
+#endif
+	return msg_data_reg;
 }
 
-#if defined(CONFIG_INTEL_VTD_ICTL) || defined(CONFIG_PCIE_MSI_X)
```

Comment:
below code should be executed also when CONFIG_MSI is set in order to make the IRW/MSI related functions available when using MSI interrupt signaling.